client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-styles.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button } from '@automattic/components';
+import { NavigatorButtonAsItem, NavigatorHeader, NavigatorItemGroup } from '@automattic/onboarding';
+import {
+	__experimentalVStack as VStack,
+	__experimentalUseNavigator as useNavigator,
+} from '@wordpress/components';
+import { color, typography } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { useRef, useEffect } from 'react';
+import { NAVIGATOR_PATHS } from './constants';
+import NavigatorTitle from './navigator-title';
+
+interface Props {
+	onMainItemSelect: ( { name, isPanel }: { name: string; isPanel?: boolean } ) => void;
+	onContinueClick: ( callback?: () => void ) => void;
+	selectedMainItem: string | null;
+	hasColor: boolean;
+	hasFont: boolean;
+}
+
+const ScreenStyles = ( {
+	onMainItemSelect,
+	onContinueClick,
+	selectedMainItem,
+	hasColor,
+	hasFont,
+}: Props ) => {
+	const translate = useTranslate();
+	const wrapperRef = useRef< HTMLDivElement | null >( null );
+	const navigator = useNavigator();
+	const headerDescription = translate(
+		'Create your homepage by first adding patterns and then choosing a color palette and font style.'
+	);
+	const getIsActiveButton = ( name: string ) => {
+		return name === selectedMainItem;
+	};
+
+	const togglePanel = () => {
+		if ( selectedMainItem ) {
+			// Toggle panel before continue
+			onMainItemSelect( { name: selectedMainItem, isPanel: true } );
+		}
+	};
+
+	const handleContinueClick = () => {
+		togglePanel();
+		onContinueClick();
+	};
+
+	useEffect( () => {
+		if ( selectedMainItem !== 'color-palettes' ) {
+			// Open Colors initially
+			setTimeout( () => onMainItemSelect( { name: 'color-palettes', isPanel: true } ), 250 );
+		}
+	}, [] );
+
+	return (
+		<>
+			<NavigatorHeader
+				title={ <NavigatorTitle title={ translate( 'Pick your style' ) } /> }
+				description={ headerDescription }
+				onBack={ () => {
+					navigator.goBack();
+					togglePanel();
+				} }
+			/>
+			<div className="screen-container__body" ref={ wrapperRef }>
+				<VStack spacing="4">
+					<NavigatorItemGroup title={ translate( 'Styles' ) }>
+						<>
+							<NavigatorButtonAsItem
+								checked={ hasColor }
+								path={ NAVIGATOR_PATHS.STYLES }
```

Comment:
@arthur791004 I'm here again using this trick to use the `NavigatorButtonAsItem` in the same path (screen). Looks a bit weird but works fine. WDYT?

Reply:
I guess it's fine for now since the current implementation needs the main item and panel